### PR TITLE
User profile data provider

### PR DESCRIPTION
**Pull request by arnifreyrs**
# \<Description\>

## What

External data provider for email and mobile number from user profile service. Use that in the parental leave form in order to skip unnecessary questions.

## Why

We should not need to ask the applicant constantly about his/her information. Let's just read it from the user profile service.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 1]
```diff
@@ -229,18 +235,23 @@ export class ApplicationController {
...
       (existingApplication as BaseApplication).typeId,
     )
+    const headers = (req.headers as unknown) as { authorization?: string }
 
     const results = await callDataProviders(
-      buildDataProviders(externalDataDto, templateDataProviders),
+      buildDataProviders(
+        externalDataDto,
+        templateDataProviders,
+        headers.authorization ?? '',
```
Comment (baering) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:253]: Is it ever an acceptable case not to have the auth header here?

File: apps/service-portal/src/components/ApplicationErrorBoundary/ApplicationErrorBoundary.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export class ApplicationErrorBoundary extends PureComponent<
   }
 
   componentDidCatch(error: Error) {
-    if (window.location.origin === 'http://localhost:4200') {
+    if (window.location.origin.includes('http://localhost')) {
```
Comment (baering) [inline apps/service-portal/src/components/ApplicationErrorBoundary/ApplicationErrorBoundary.tsx:30]: We should probably also [disable Sentry](https://github.com/island-is/island.is/blob/main/apps/application-system/form/src/main.tsx#L9) if not running in a production environment

File: apps/application-system/api/src/app/modules/application/application.controller.ts (hunk 1 above)
Comment (arnifreyrs) [reply apps/application-system/api/src/app/modules/application/application.controller.ts:253]: yes some data providers will not need to use it. I am not sure that the data providers that call the parental leave API will use it actually,

File: libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts [hunk 3]
```diff
@@ -57,6 +57,13 @@ export const ParentalLeaveForm: Form = buildForm({
               name: m.introductionProvider,
               id: 'approveExternalData',
               dataProviders: [
+                buildDataProviderItem({
+                  id: 'userProfile',
+                  type: 'UserProfileProvider',
+                  title: 'Your user profile information',
```
Comment (baering) [inline libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts:63]: Let's place this in a translation file.

File: libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts [hunk 4]
```diff
@@ -57,6 +57,13 @@ export const ParentalLeaveForm: Form = buildForm({
...
+                  subTitle:
+                    'In order to apply for this application we need your email and phone number',
```
Comment (baering) [inline libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts:65]: Let's place this in a translation file.

**Review by baering — DISMISSED** [review]

**Review by Thorrik58 — APPROVED** [review]
Looks good on our end 👍

**Review by Thorrik58 — APPROVED** [review]

**Review by baering — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]
LGTM

**Pull request merged**